pkg/azurecluster/validate_create.go

```diff
@@ -58,7 +58,14 @@ func (a *CreateValidator) Validate(ctx context.Context, request *v1beta1.Admissi
 		return microerror.Maskf(errors.ParsingFailedError, "unable to parse AzureCluster CR: %v", err)
 	}
 
-	err := generic.ValidateOrganizationLabelContainsExistingOrganization(ctx, a.ctrlClient, azureClusterCR)
+	err := azureClusterCR.ValidateCreate()
+	if CAPZIsNameFieldValidationError(err) {
```

Comment:
This validation error came up when using clusters that had uppercase characters in their names. Let's skip it for now.

Reply:
what if we don't ignore and add that validation? I don't think we use uppercase characters in cluster names anyway. What would that imply? Updating docs? WDYT?

Reply:
Turns out our generated IDs don't have uppercase characters anyway (https://github.com/giantswarm/kubectl-gs/blob/2543fd3ba51cefb5494a1fdde43b148fdff334a2/internal/key/key.go#L29 and https://github.com/giantswarm/api/blob/8ff9eb0d8f25ebe65abe390eb7b0f8e38406425b/pkg/entityid/generator.go#L26). It was an error that only occured in tests. I'll update the code

Reply:
Ah wait, the name must not start or end with a number, which our generated IDs sometimes do. I need to update that ID generation

Reply:
Oh you are right about starting with a number, but I think it allos ending with a number.

Reply:
It allows ending with a number, that's true. But we have a problem. All the existing clusters that have an ID that starts with a number will fail to validate if anyone tries to update them.

The error message says that it can begin with a alphanumeric character (https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/417eccbe0ca8a433f44d1fb2a3a473c494a5ada3/api/v1alpha3/azurecluster_validation.go#L78), but that isn't true, since it can't begin with a number. 

Is the validation wrong, or the error message? 😅

Reply:
Let's ignore that error for now so our clusters won't fail, and bring this upstream. WDYT?